demos/custom-tabs-authview/src/main/java/com/google/androidbrowserhelper/demos/customtabsauthview/AuthManager.java

```diff
@@ -0,0 +1,99 @@
+package com.google.androidbrowserhelper.demos.customtabsauthview;
+
+import android.content.Context;
+import android.content.Intent;
+import android.content.SharedPreferences;
+import android.net.Uri;
+import android.os.Handler;
+import android.os.Looper;
+import android.util.Log;
+
+import androidx.activity.result.ActivityResultLauncher;
+import androidx.annotation.NonNull;
+import androidx.browser.auth.AuthTabIntent;
+import androidx.annotation.OptIn;
+import androidx.browser.auth.ExperimentalAuthTab;
+
+import java.io.IOException;
+import java.util.UUID;
+
+@OptIn(markerClass = ExperimentalAuthTab.class)
+public class AuthManager {
+    private static final String TAG = "OAuthManager";
+
+    private final String mClientId;
+    private final String mClientSecret;
+    private final String mAuthorizationEndpoint;
+    private final String mRedirectScheme;
+
+    public interface OAuthCallback {
+        void auth(String accessToken, String scope, String tokenType);
+    }
+
+    public AuthManager(String clientId, String clientSecret, String authorizationEndpoint,
+                       String redirectScheme) {
+        mClientId = clientId;
+        mClientSecret = clientSecret;
+        mAuthorizationEndpoint = authorizationEndpoint;
+        mRedirectScheme = redirectScheme;
+    }
+
+    public void authorize(Context context, ActivityResultLauncher<Intent> launcher, String scope) {
+        // Generate a random state.
+        String state = UUID.randomUUID().toString();
+
+        // Save the state so we can verify later.
+        SharedPreferences preferences =
+                context.getSharedPreferences("OAUTH_STORAGE", Context.MODE_PRIVATE);
+        preferences.edit()
+                .putString("OAUTH_STATE", state)
+                .apply();
+
+        // Create an authorization URI to the OAuth Endpoint.
+        Uri uri = Uri.parse(mAuthorizationEndpoint)
+                .buildUpon()
+                .appendQueryParameter("response_type", "code")
+                .appendQueryParameter("client_id", mClientId)
+                .appendQueryParameter("scope", scope)
+                .appendQueryParameter("state", state)
+                .build();
+
+        // Open the Authorization URI in a Chrome Custom Auth Tab.
+        AuthTabIntent authTabIntent = new AuthTabIntent.Builder().build();
```

Comment:
I see that AuthView demo is almost identical to the Oauth demo, the difference are:
1. New callback intent-filter
2. Launcher intents are different
3. Different result handling

I think that we can merge this demo into Oauth, as far as I can see it'd require:
1.  Add an AuthView callback intent-filter into the AndroidManifest
2. Demo should provide a way to switch between CCT and AuthView. I think a checkbox and `shouldUseAuthView` should be enough.
3. In AuthManager launch intents and handle results based on `shouldUseAuthView`.
4. I'd also add comments why AuthView is preferable in AuthManager or even deprecate CCT flow at all.

This way we get rid of duplication and provide some kind of migration steps from CCT to AuthView.
What do you think?

Reply:
Hi @ukaratkevich, 

You're right that these two demos are very similar. The new Auth Tab implementation is intended to be a replacement for the older Oauth implementation strategy. The Oauth project uses standard CCT. The new project uses Auth Tab but also leverages the older CCT method as a fallback. So there are a lot of pieces between the two demos that are identical. I was asked not to delete the Oauth project but instead to keep it for historical purposes. 

As far as what to do next, it's important to know that Auth Tab *is* CCT, and so we shouldn't really be representing a choice between one or the other to the developer. Instead, the demo shows the new Auth Tab implementation with the standard CCT as a graceful fallback, which is I think is the flow we want to demonstrate to developers. Adding booleans and checkboxes defeats that intention. 

I think perhaps it would be easiest to delete the older Oauth demo entirely, if that is creating confusion. Would that help?

Reply:
Thanks a lot for the explanation, in this case I'd not delete the oath demo, but rather mark it as deprecated in the MD file with `@Deprecated` annotation on main classes and add a comment to use Auth View instead.

Overall, lgtm :)